### PR TITLE
Fix: Update nested list spacing

### LINKED ISSUE #1359: List: Spacing issue in nested list
There should be a `spacing-2` between the elements here: 

Image

### PR DESCRIPTION
**Pull request by ramanand-rv**
This PR aims to fix #1359 

### Changes Made:

Adjusted styles for the nested List and List Item components.
Set margin-top for the nested List Item to var(--fds-spacing-2).

Please review the changes in the styles for the nested List and List Item components. Ensure that the spacing adjustments align with the design requirements. The goal is to enhance the visual presentation and maintain a clear hierarchy within the nested list structure. In case of further changes please notify me, I would love to contribute here.

Thank you for your review!

### CODE REVIEW

**Review by mimarz — CHANGES_REQUESTED** [review]
Hi @ramanand-rv!

Thanks for wanting to contribute to our project! I left you a review on respective changes in code as a separate comment.

In addition:
- Make sure to have the correct line ending (LF) for the file. You can see Github giving a warning in the "files changed" tab.
- We use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/) and use scope for the affected components, which means the PR title should start with "fix(List):" instead of just "fix:"

**Comment by mrosvik** [comment]
@ramanand-rv Thank you for contributing ❤️ And I apologize for my poor explanation. I meant to say that the _Nested_ list was missing spacing-2 above it, where the red arrow is. 😊

File: packages/react/src/components/List/List.module.css [hunk 1]
```diff
@@ -9,3 +9,7 @@
 li > .list {
   margin-top: var(--fds-spacing-2);
 }
+
+.list .listItem {
+  margin-top: var(--fds-spacing-2);
+}
```
[the remark this answers is not in the document]
Comment (ramanand-rv) [reply packages/react/src/components/List/List.module.css:15]: Thanks @mimarz for the helpful review comment. Will implement the changes and update the PR accordingly.

**Comment by Barsnes** [comment]
Hi @ramanand-rv, do you have a time estimate of when you would be able to get this done?

**Comment by Barsnes** [comment]
Closing this issue as we waited a week with no changes made.
This will be resolved in #1402

**Pull request closed without merging**